modules/ide/xsk-ide-xsaccess/src/main/resources/META-INF/dirigible/xsk-ide-xsaccess/editor/editor.js

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+angular.module('page', [])
+  .controller('PageController', function ($scope) {
+
+    let messageHub = new FramesMessageHub();
+    let contents;
+
+    function getResource(thePath) {
+      let xhr = new XMLHttpRequest();
+      xhr.open('GET', thePath, false);
+      xhr.send();
+      if (xhr.status === 200) {
+        return xhr.responseText;
+      }
+    }
+
+    function loadContents(xsaccessfile) {
+      if (!xsaccessfile) {
+        console.error('XSACCESS file parameter is not present in the URL');
+      } else {
+        return getResource('/services/v4/ide/workspaces' + xsaccessfile);
```

Comment:
Better add a return to the `if` statement above and just `return` here. That way you wouldn't need the `else` branch and there would be less nesting in the code

Reply:
OK